scripts/debug/collect-logs.sh

```diff
@@ -0,0 +1,81 @@
+
+CLUSTER_NAME=${CLUSTER_NAME:-"test-$(date +%s)"}
+ARTIFACTS="${ARTIFACTS:-${PWD}/_artifacts}"
+
+# dump logs from kind and all the nodes
+dump-logs() {
+  # log version information
+  echo "=== versions ==="
+  echo "kind : $(kind version)" || true
+  echo "kubectl: "
+  kubectl --kubeconfig=$CLUSTER_NAME.kubeconfig version || true
+  echo ""
+
+  # dump all the info from the CAPI related CRDs
+  mkdir -p $ARTIFACTS/logs
+  kubectl get \
+  clusters,azureclusters,machines,azuremachines,kubeadmconfigs,machinedeployments,azuremachinetemplates,kubeadmconfigtemplates,machinesets,kubeadmcontrolplanes \
+  --all-namespaces -o yaml >> "${ARTIFACTS}/logs/capz.info" || true
+
+  # dump images info
+  echo "images in docker" >> "${ARTIFACTS}/logs/images.info"
+  docker images >> "${ARTIFACTS}/logs/images.info"
+  echo "images from bootstrap using containerd CLI" >> "${ARTIFACTS}/logs/images.info"
+  docker exec kind-control-plane ctr -n k8s.io images list >> "${ARTIFACTS}/logs/images.info" || true #TODO: make this work for kind or tilt?
+  echo "images in bootstrap cluster using kubectl CLI" >> "${ARTIFACTS}/logs/images.info"
+  (kubectl get pods --all-namespaces -o json \
+   | jq --raw-output '.items[].spec.containers[].image' | sort)  >> "${ARTIFACTS}/logs/images.info" || true
+  echo "images in deployed cluster using kubectl CLI" >> "${ARTIFACTS}/logs/images.info"
+  (kubectl --kubeconfig=$CLUSTER_NAME.kubeconfig get pods --all-namespaces -o json \
+   | jq --raw-output '.items[].spec.containers[].image' | sort)  >> "${ARTIFACTS}/logs/images.info" || true
+
+  # dump cluster info for kind
+  kubectl cluster-info dump > "${ARTIFACTS}/logs/kind-cluster.info" || true
+
+  # dump cluster info for capz
+  echo "VMs owned by CAPZ in resource group ${AZURE_RESOURCE_GROUP}  ===" >> "${ARTIFACTS}/logs/capz-cluster.info" || true
+ az vm list --resource-group "${AZURE_RESOURCE_GROUP}" --query "[?tags.\"sigs.k8s.io_cluster-api-provider-azure_cluster_capi-quickstart\" == 'owned']" >> "${ARTIFACTS}/logs/capz-cluster.info" || true
+
+
+  echo "=== cluster-info dump ===" >> "${ARTIFACTS}/logs/capz-cluster.info" || true
+  kubectl --kubeconfig=$CLUSTER_NAME.kubeconfig cluster-info dump >> "${ARTIFACTS}/logs/capz-cluster.info" || true
+
+  # export all logs from kind
+  kind "export" logs --name="kind" "${ARTIFACTS}/logs" || true
+
+
+  # TODO: Convert below from aws to azure (WIP)
+  node_filter="Name=tag:sigs.k8s.io/cluster-api-provider-azure/cluster/${CLUSTER_NAME},Values=owned"
+  bastion_filter="Name=tag:sigs.k8s.io/cluster-api-provider-azure/role,Values=bastion"
+  jump_node=$(aws ec2 describe-instances --region "$AWS_REGION" --filters "${node_filter}" "${bastion_filter}" --query "Reservations[*].Instances[*].PublicIpAddress" --output text | head -1)
```

Comment:
I'm guessing you will be changing this over to `az`? If not, you could probably just query the instance metadata endpoint to get most of the data.

Reply:
I spent a painfully long time crafting the jmespath query so I'm using that. Can switch to instance metadata endpoint if you want but that would require running the command on a vm which seems like it could be less straightforward

Reply:
Works for me!